server/request_record.go

```diff
@@ -0,0 +1,71 @@
+package server
+
+import (
+	"fmt"
+	"github.com/flashbots/rpc-endpoint/database"
+	"github.com/google/uuid"
+	"net/http"
+	"sync"
+)
+
+type requestRecord struct {
+	requestEntry        *database.RequestEntry
+	ethSendRawTxEntries []*database.EthSendRawTxEntry
+	mutex               *sync.Mutex
+	db                  database.Store
```

Comment:
shouldn't this be a pointer to the db store?

Reply:
Its an interface. Not sure what do you mean ?

Reply:
How is the actual database instance passed in here, by value or by reference?

Reply:
by reference